Crud.php

```diff
@@ -0,0 +1,271 @@
+         */
+        public function readAllNumbers():object {
+            $numberRepository = $this->entityManager->getRepository('Number');
+            $numbers = $numberRepository->findAll();
+
+            return $numbers;
+        }
+
+        /**
+         * Reads numbers paginated
+         * 
+         * @param int page
+         * 
+         * @return array
+         */
+        public function readNumbersPaginate(int $page = 1):array {               
+            // get the user repository
+            $numbers =  $this->entityManager->getRepository(Number::class);
+
+            // build the query for the doctrine paginator
+            $query = $numbers->createQueryBuilder('n')
+                ->getQuery();
+
+            //set page size
+            $pageSize = self::PER_PAGE;
+
+            // load doctrine Paginator
+            $paginator = new \Doctrine\ORM\Tools\Pagination\Paginator($query);
+
+            // you can get total items
+            $totalItems = count($paginator);
+
+            // get total pages
+            $pagesCount = ceil($totalItems / $pageSize);
+
+            // now get one page's items:
+            $paginator
+                ->getQuery()
+                ->setFirstResult($pageSize * ($page-1)) // set the offset
+                ->setMaxResults($pageSize); // set the limit
+
+            $result = [
+                'page' => $page,
+                'data' => $paginator,
+                'pages' => $pagesCount
+            ];
+
+            return $result;
+        }
+
+        /**
+         * Reads number with seted filters and sortings parametters without pagination
+         * 
+         * @param int page
+         * @param array filters
+         * @param array sortings
+         * 
+         * @return array
+         */
+        public function readNumbersWithFilterAndSortings(int $page = 1, array $filters = [], array $sortings = []):array { 
+            $numbers =  $this->entityManager->getRepository(Number::class)->findBy($filters, $sortings);
+
+            $result = [
+                'page' => 1,
+                'data' => $numbers,
+                'pages' => 1
+            ];
+
+            return $result;
+        }
+
+        /**
+         * Read numbers with pagination, sorting and filtering
+         * 
+         * @param int page
+         * @param array filters
+         * @param string sorting
+         * 
+         * @return array
+         */
+        public function readNumbers(int $page = 1, array $filters = [], string $sorting = "ASC") {               
+            // get the user repository
+            $numbers =  $this->entityManager->getRepository(Number::class);
+
+            // build the query for the doctrine paginator
+            $query = $numbers->createQueryBuilder('n');
+
+            //setting sorting
+            $query = $query->orderBy('n.slug', $sorting);
+
+            //Adding searching parametters
+            if(!empty($filters['slug'])) {
```

Comment:
Smells like code duplication line 137 and line 190. Please try to refactor it

Reply:
Will refactor read number paginated